internal/hcl/evaluator.go

```diff
@@ -516,13 +517,27 @@ func (e *Evaluator) expandBlockForEaches(blocks Blocks) Blocks {
 				cloneValues := clone.Values()
 				e.ctx.Set(cloneValues, append(labels, keyStr)...)
 
-				if v, ok := e.moduleCalls[clone.FullName()]; ok {
-					v.Definition = clone
+				if clone.Type() == "module" {
+					if v, ok := e.moduleCalls[clone.FullName()]; ok {
+						v.Definition = clone
+					} else {
+						modCall, err := e.loadModule(clone)
+						if err != nil {
+							e.logger.WithError(err).Debugf("could not load module %s", clone.FullName())
+							return false
+						}
+						e.moduleCalls[clone.FullName()] = modCall
+					}
 				}
+
 				expanded = append(expanded, clone)
 
 				return false
 			})
+
+			if block.Type() == "module" {
+				delete(e.moduleCalls, block.FullName())
```

Comment:
can we add a log line here just to say which module we're deleting from the calls.